### PR TITLE
create variations manually/asynchronously

### PR DESCRIPTION
**Pull request by anneFly**

### CODE REVIEW

File: stdimage/models.py [hunk 1]
```diff
@@ -36,18 +36,26 @@ class StdImageFieldFile(ImageFieldFile):
...
             or img.size[1] > variation['height']
 
-    def render_and_save_variation(self, name, content, variation,
+    def render_and_save_all_variations(self):
+        """
+        Renders all image variations and saves them to the storage
+        """
+        variations = self.field.variations
+        for key, variation in variations.items():
+            self.render_and_save_variation(self, variation)
```
Comment (codingjoe) [inline stdimage/models.py]: the content argument is missing

File: stdimage/models.py [hunk 2]
```diff
@@ -148,7 +156,8 @@ class StdImageField(ImageField):
     }
 
     def __init__(self, verbose_name=None, name=None, variations=None,
-                 force_min_size=False, *args, **kwargs):
+                 create_variations_manually=False, force_min_size=False,
```
Comment (codingjoe) [inline stdimage/models.py]: I still prefer a callable that is called when the image is saved to the storage.

File: tests/models.py [hunk 3]
```diff
@@ -2,8 +2,9 @@
 from django.db import models
 
 from stdimage import StdImageField
-from stdimage.utils import pre_delete_delete_callback, pre_save_delete_callback, \
-    UploadTo, UploadToAutoSlugClassNameDir, UploadToUUID
+from stdimage.utils import pre_delete_delete_callback, \
+    pre_save_delete_callback, UploadTo, UploadToAutoSlugClassNameDir, \
+    UploadToUUID
 from stdimage.validators import MaxSizeValidator, MinSizeValidator
```
Comment (codingjoe) [inline tests/models.py]: tests are missing

**Comment by codingjoe** [comment]
This feature needs some serious testing and documentation.
Also I'd like to have a callback that triggers a jobs to best started to ensure that the file is on the storage and the db transaction is committed.

File: stdimage/models.py (hunk 1 above)
Comment (anneFly) [inline stdimage/models.py]: self is the content

**Comment by anneFly** [comment]
I added a test and small fixes.
I did not refactor to using a callable, yet since all my tries on this failed somehow. Either you take over on this or we could do it together.

**Comment by codingjoe** [comment]
@anneFly I will do some further work on this and merge eventually

**Comment by codingjoe** [comment]
I did some changed and merged in 7d8d244c765376394a6c42dc49853df1e8a766a0

**Pull request closed without merging**